NangoHQ/nango
# feat: reconnect

## Changes

Fixes https://linear.app/nango/issue/NAN-2302/connect-refuse-connectionid-when-auth-is-session-token
Fixes https://linear.app/nango/issue/NAN-2249/connect-reconnect

- Ability to create a session token to reconnect a `connection_id`
After debate with the team, we settled with a new endpoint dedicated to creating a reconnect token.

- Refuse `connection_id` has a query params if a session token is used

- Update all endpoints to add the reconnect ability

- Add SDK, doc, openapi 

## 🧪 How to test?

There is no built-in way to reconnect from the UI so it's a bit manual

- Use your favorite curl/api tool to create a token from 
```sh
curl --request POST \
  --url http://localhost:3003/connect/sessions/reconnect \
  --header 'Authorization: _PRIVATE_KEY_' \
  --header 'Content-Type: application/json' \
  --data '{
	"connection_id": "test-connection-id",
	"integration_id": "google-mail",
  "end_user": {
    "id": "test_insomnia",
    "email": "samuel@nango.dev"
  },
}'
```
- Access Connect UI directly with the token `http://localhost:3009/?session_token=SESSION_TOKEN`
- Connect
- Check that no connection has been created and the selected `connection_id` has been updated
